server/src/main/java/org/elasticsearch/index/query/QueryShardContext.java

```diff
@@ -290,8 +292,9 @@ MappedFieldType failIfFieldMappingNotFound(String name, MappedFieldType fieldMap
 
     public SearchLookup lookup() {
         if (lookup == null) {
-            lookup = new SearchLookup(getMapperService(),
-                    mappedFieldType -> indexFieldDataService.apply(mappedFieldType, fullyQualifiedIndex.getName()));
+            lookup = new SearchLookup(
+                getMapperService(),
+                (fieldType, lookup) -> indexFieldDataService.apply(fieldType, fullyQualifiedIndex.getName(), lookup));
```

Comment:
Shadowing the `lookup` name here makes this harder to read. I probably had this on my version of the PR too, but could you rename `lookup` to `l` or something?

Reply:
makes sense